app/database/data_store.go

```diff
@@ -222,6 +231,14 @@ func (s *DataStore) InTransaction(txFunc func(*DataStore) error) error {
 	return err
 }
 
+// SchedulePropagation schedules a run of the propagation for the given types after the transaction commit.
+func (s *DataStore) SchedulePropagation(types []string) {
+	s.mustBeInTransaction()
+
+	triggersToRun := s.DB.ctx.Value(triggersContextKey).(*awaitingTriggers)
+	triggersToRun.SchedulePropagationTypes = utils.UniqueStrings(append(triggersToRun.SchedulePropagationTypes, types...))
```

Comment:
If the order of types doesn't matter, It would be much clearer to store map[string]struct{} in triggersToRun.SchedulePropagationTypes instead of creating utils.UniqueStrings() and tests for it.

If the order of types matters, UniqueString() breaks it anyway since map doesn't preserve the order of keys.